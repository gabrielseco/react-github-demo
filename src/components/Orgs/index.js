export { default as Orgs } from './Orgs';
