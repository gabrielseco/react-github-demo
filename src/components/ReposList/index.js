export { default as ReposList } from './ReposList';
