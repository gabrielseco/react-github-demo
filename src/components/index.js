export * from './Header';
export * from './Footer';
export * from './NavBar';
export * from './NavItem';
export * from './SearchBox';
export * from './Spinner';
