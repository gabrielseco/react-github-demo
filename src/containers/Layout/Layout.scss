.wrapper {
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
}

.searchBoxContainer {
  margin-top: 5%;
}

.searchBox {
  padding: 10px;
  width: 300px;
}
