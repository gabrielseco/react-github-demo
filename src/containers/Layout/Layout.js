import React from 'react';
import { Header, Footer, SearchBox } from './../../components';
import styles from './Layout.scss';
import utils from './../../stylesheets/utils/flexbox.scss';

export default function Layout({ routes, onChange }) {
  return (
    <div className={utils.flexboxSticky}>
      <Header />
      <main className={utils.flex1}>
        <div className={styles.wrapper}>
          <div className={styles.searchBoxContainer}>
            <SearchBox
              classesInput={styles.searchBox}
              placeholder="Start typing to search for users"
              onChange={onChange}
            />
          </div>
        </div>
        {routes}
      </main>
      <Footer />
    </div>
  );
}
