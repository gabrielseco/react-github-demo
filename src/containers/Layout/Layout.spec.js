import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import { ReposPage } from './../../components';

import Layout from './Layout';

describe('Layout suite', () => {
  it('renders Layout', () => {
    const routes = (
      <Switch>
        <Route path="/" component={ReposPage} />
      </Switch>
    );
    const comp = (
      <Router>
        <Layout routes={routes} onChange={() => {}} />
      </Router>
    );
    const component = shallow(comp);
    const tree = renderer.create(comp).toJSON();
    expect(component).toBeDefined();
    expect(tree).toMatchSnapshot();
  });
});
