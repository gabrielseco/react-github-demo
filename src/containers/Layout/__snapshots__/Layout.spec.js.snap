// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout suite renders Layout 1`] = `
<div
  className="flexboxSticky"
>
  <header
    className="header"
  >
    <div
      className="logo"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <h2
          className="title"
        >
          Github Demo
        </h2>
      </a>
    </div>
    <nav
      className="navBar"
    >
      <ul
        className="navWrapper"
      >
        <li
          className="navItem"
        >
          <a
            className="navLink"
            href="/"
            onClick={[Function]}
          >
            Repos
          </a>
        </li>
        <li
          className="navItem"
        >
          <a
            className="navLink"
            href="/orgs"
            onClick={[Function]}
          >
            Orgs
          </a>
        </li>
      </ul>
    </nav>
  </header>
  <main
    className="flex1"
  >
    <div
      className="wrapper"
    >
      <div
        className="searchBoxContainer"
      >
        <div
          className="container"
        >
          <input
            className="searchBox"
            onKeyUp={[Function]}
            placeholder="Search Users"
            type="text"
          />
        </div>
      </div>
    </div>
  </main>
  <footer
    className="footer"
  >
    <p>
      Gabriel Garc√≠a Seco
    </p>
    <p>
      Made with 
      <i
        className="footerIcon fa fa-heart-o"
      />
      , Madrid 2018
    </p>
  </footer>
</div>
`;
